doc/includes/API.md

```diff
@@ -4968,10 +4968,16 @@ case of ManyToManyRelation also the join table needs to be defined. This is done
 The `modelClass` passed to the relation mappings is the class of the related model. It can be one of the following:
 
 1. A model class constructor
-2. An absolute path to a module that exports a model class
-3. A path relative to one of the paths in [`modelPaths`](#modelpaths) array.
+1. A thunk (a function with no arguments) that returns a model class constructor
+1. An absolute path to a module that exports a model class
+1. A path relative to one of the paths in [`modelPaths`](#modelpaths) array.
+
+The thunk and file path versions avoid require loops.
 
-The file path versions are handy for avoiding require loops.
+Thunks are recommended if you are packing up your server-side code, as the
```

Comment:
After successfully using thunks in my own project, I'd suggest we make this the default recommendation. `() => Movie` is straightforward, doesn't cause `webpack` or similar tools heartache due to synthetic `require`s, and avoids circular require loops (whose errors are always confusing). I'll let you decide how much you want to push this, of course.

Reply:
Just to add to this:
I have a habit of memoizing at the top of file, e.g.

```js
const { memoize } = require('lodash')
const MyModel = memoize(() => require('./relative/path/to'))

// ...
await Model().query()
```
Sort of helps normalise model dependency throughout the application, and also help with re/up-start time as projects grows. Memoizing was more relevant pre node 8 (or so) as `require` wasn't really optimised at all. Habits are tough to shake.